### PR TITLE
Change GetCoinInfo and add test for it

### PR DESCRIPTION
**Pull request by EnoRage**

### CODE REVIEW

File: services/assets/info.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func (cl *HttpAssetClient) GetCoinInfo(coinId int, token string) (*watchmarket.C
 
 	err = json.Unmarshal(resp.Body(), &info)
 	if err != nil {
-		return nil, err
+		return &info, errors.New(fmt.Sprintf("Unmarshal %s", err.Error()))
```
Comment (arein) [inline services/assets/info.go]: minor/nonblocking: "Failed to unmarshal" would be more descriptive.

File: services/assets/info.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ func (cl *HttpAssetClient) GetCoinInfo(coinId int, token string) (*watchmarket.C
 
 	err = json.Unmarshal(resp.Body(), &info)
```
Comment (arein) [inline services/assets/info.go]: can you rebase fix/tests? On [master there is a `TODO`](https://github.com/trustwallet/watchmarket/blob/master/services/assets/info.go#L49) that we don't need anymore now

Comment (EnoRage) [reply services/assets/info.go]: Yep

File: services/assets/info.go (hunk 1 above)
Comment (EnoRage) [reply services/assets/info.go]: Ok

File: api/api_test.go [hunk 3]
```diff
@@ -252,6 +252,22 @@ func TestCoinInfo(t *testing.T) {
...
+			expectedBody:   "{\"code\":404,\"error\":\"Coin info for coin id 500 (token: ETHToken, currency: USD) not found\"}",
+		},
+		{
+			name:           "test coin assets not found",
+			requestUrl:     fmt.Sprintf("%s/v1/market/info?coin=1000&token=ETHToken", server.URL),
+			requestMethod:  "GET",
+			requestBody:    "",
+			expectedStatus: 200,
+			expectedBody:   `{"circulating_supply": 0, "market_cap": 0, "total_supply": 0, "volume_24": 0}`,
+		},
```
Comment (arein) [inline api/api_test.go]: this shouldn't be part of this review. [This is already on master](https://github.com/trustwallet/watchmarket/blob/master/api/api_test.go#L255-L270). Can you please rebase before pushing?

File: api/marketdata.go [hunk 4]
```diff
@@ -206,6 +206,7 @@ func getCoinInfoHandler(charts *market.Charts, ac assets.AssetClient) func(c *gi
 		token := c.Query("token")
 		currency := c.DefaultQuery("currency", watchmarket.DefaultCurrency)
 		chart, err := charts.GetCoinInfo(uint(coinId), token, currency)
+		// TODO: cover special casing of TWT token in tests
```
Comment (arein) [inline api/marketdata.go]: also already on master. pls rebase

File: api/marketdata.go [hunk 5]
```diff
@@ -217,7 +218,7 @@ func getCoinInfoHandler(charts *market.Charts, ac assets.AssetClient) func(c *gi
 
 		chart.Info, err = ac.GetCoinInfo(coinId, token)
 		if err == watchmarket.ErrNotFound {
-			logger.Error(err, fmt.Sprintf("Coin assets for coin id %d (token: %s) not found", coinId, token))
+			logger.Warn(err, fmt.Sprintf("Coin assets for coin id %d (token: %s) not found", coinId, token))
```
Comment (arein) [inline api/marketdata.go]: needs rebase

**Pull request closed without merging**